algorithms/unit_tests/TestBinSortA.hpp

```diff
@@ -266,5 +266,19 @@ TEST(TEST_CATEGORY, BinSortEmptyKeysView) {
   ASSERT_NO_THROW(Sorter.create_permute_vector(ExecutionSpace{}));
 }
 
+TEST(TEST_CATEGORY, BinSort_issue_7221) {
```

Comment:
Any way you could add some comment(s) so the reader knows what this test is for and how it exposes that bug we add.

Reply:
Please add a comment to make it clear that this test is intended to exert the case where we "sort within bins" whatever that means.